src/sources/file.rs

```diff
@@ -284,6 +291,10 @@ fn default_line_delimiter() -> String {
     "\n".to_string()
 }
 
+const fn default_rotate_wait_sec() -> Duration {
+    Duration::from_secs(u64::MAX/2)
+}
```

Comment:
Small nit - we can use `Duration::MAX` here.
```suggestion
const fn default_rotate_wait() -> Duration {
    Duration::MAX
}
```

Reply:
That was my first instict, too, to use Duration::MAX, but I'm having trouble getting it through Serde. There're tests failing because of an overflow:
```
--- TRY 4 STDERR:        vector sources::file::tests::generate_config ---
thread 'sources::file::tests::generate_config' panicked at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde_with-3.4.0/src/utils/duration.rs:142:17:
attempt to add with overflow
stack backtrace:
   0:     0x561cf904ac3c - std::backtrace_rs::backtrace::libunwind::trace::ha637c64ce894333a
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/../../backtrace/src/backtrace/libunwind.rs:104:5
   1:     0x561cf904ac3c - std::backtrace_rs::backtrace::trace_unsynchronized::h47f62dea28e0c88d
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x561cf904ac3c - std::sys_common::backtrace::_print_fmt::h9eef0abe20ede486
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x561cf904ac3c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hed7f999df88cc644
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x561cf909b4b0 - core::fmt::rt::Argument::fmt::h1539a9308b8d058d
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/fmt/rt.rs:142:9
   5:     0x561cf909b4b0 - core::fmt::write::h3a39390d8560d9c9
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/fmt/mod.rs:1120:17
   6:     0x561cf904029f - std::io::Write::write_fmt::h5fc9997dfe05f882
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/io/mod.rs:1762:15
   7:     0x561cf904aa24 - std::sys_common::backtrace::_print::h894006fb5c6f3d45
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x561cf904aa24 - std::sys_common::backtrace::print::h23a2d212c6fff936
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x561cf904d6b7 - std::panicking::default_hook::{{closure}}::h8a1d2ee00185001a
  10:     0x561cf904d41f - std::panicking::default_hook::h6038f2eba384e475
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:292:9
  11:     0x561cf904dbf8 - std::panicking::rust_panic_with_hook::h2b5517d590cab22e
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:779:13
  12:     0x561cf904daa9 - std::panicking::begin_panic_handler::{{closure}}::h233112c06e0ef43e
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:649:13
  13:     0x561cf904b106 - std::sys_common::backtrace::__rust_end_short_backtrace::h6e893f24d7ebbff8
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/sys_common/backtrace.rs:170:18
  14:     0x561cf904d842 - rust_begin_unwind
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:645:5
  15:     0x561ce8018d55 - core::panicking::panic_fmt::hbf0e066aabfa482c
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panicking.rs:72:14
  16:     0x561ce8018df3 - core::panicking::panic::h195fc2a96298d4c3
                               at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panicking.rs:127:5
  17:     0x561cece433bc - serde_with::utils::duration::<impl serde_with::ser::SerializeAs<serde_with::utils::duration::DurationSigned> for serde_with::DurationSeconds<u64,STRICTNESS>>::serialize_as::h76cd3c5390cbf2e4
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde_with-3.4.0/src/utils/duration.rs:142:17
  18:     0x561cece4273a - serde_with::ser::impls::<impl serde_with::ser::SerializeAs<core::time::Duration> for serde_with::DurationSeconds<u64,STRICTNESS>>::serialize_as::h2e8a6e3f02bb49b2
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde_with-3.4.0/src/ser/impls.rs:623:21
  19:     0x561cece42605 - serde_with::ser::<impl serde_with::As<T>>::serialize::h7b3e747688c698ae
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde_with-3.4.0/src/ser/mod.rs:171:9
  20:     0x561cec8a9118 - <vector::sources::file::_::<impl serde::ser::Serialize for vector::sources::file::FileConfig>::serialize::__SerializeWith as serde::ser::Serialize>::serialize::h1f0ef1d52760709b
                               at /home/syedriko/rust/upstream/vector/src/sources/file.rs:75:1
  21:     0x561cec898608 - serde::ser::impls::<impl serde::ser::Serialize for &T>::serialize::h4be1b9aee57ba086
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.195/src/ser/impls.rs:500:17
  22:     0x561ceb96db8b - toml::value::Value::try_from::h2de305dca759737e
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/toml-0.8.8/src/value.rs:52:9
  23:     0x561ceb9d28ae - <toml::value::SerializeMap as serde::ser::SerializeMap>::serialize_value::h7af1b143e6e991dc
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/toml-0.8.8/src/value.rs:1333:15
  24:     0x561ceb9e75ca - <toml::value::ValueSerializeMap as serde::ser::SerializeMap>::serialize_value::h43d35a97ec47fa59
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/toml-0.8.8/src/value.rs:1390:9
  25:     0x561ceb9f257e - <toml::value::ValueSerializeMap as serde::ser::SerializeStruct>::serialize_field::hc2718211e157fa11
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/toml-0.8.8/src/value.rs:1411:9
  26:     0x561cec8a78ea - vector::sources::file::_::<impl serde::ser::Serialize for vector::sources::file::FileConfig>::serialize::hdb03f2a9daac2024
                               at /home/syedriko/rust/upstream/vector/src/sources/file.rs:79:12
  27:     0x561cec898698 - serde::ser::impls::<impl serde::ser::Serialize for &T>::serialize::h8bc8114217a0b87f
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.195/src/ser/impls.rs:500:17
  28:     0x561ceb971fcb - toml::value::Value::try_from::hc02990178f52cb58
                               at /home/syedriko/.cargo/registry/src/index.crates.io-6f17d22bba15001f/toml-0.8.8/src/value.rs:52:9
  29:     0x561cecae57fe - <vector::sources::file::FileConfig as vector_config::component::generate::GenerateConfig>::generate_config::hf06adc827a9bdaf4
                               at /home/syedriko/rust/upstream/vector/lib/vector-config/src/component/generate.rs:11:17
  30:     0x561ce8214f66 - vector::test_util::test_generate_config::h7c01ac7fb208a397
                               at /home/syedriko/rust/upstream/vector/src/test_util/mod.rs:99:32
  31:     0x561ced7e20a6 - vector::sources::file::tests::generate_config::h7a4d234e6412afb1
                               at /home/syedriko/rust/upstream/vector/src/sources/file.rs:869:9
  32:     0x561ce9122a47 - vector::sources::file::tests::generate_config::{{closure}}::hca693cb67401a2dc
                               at /home/syedriko/rust/upstream/vector/src/sources/file.rs:868:25
```
`u64::MAX/2` is a pretty large number for it to be practical infinity.